lib/streamlit/elements/dialog_decorator.py

```diff
@@ -94,56 +94,80 @@ def dialog_decorator(title: F | None, *, width: DialogWidth = "small") -> F:
 def dialog_decorator(
     title: F | None | str = "", *, width: DialogWidth = "small"
 ) -> F | Callable[[F], F]:
-    r"""Decorate a function to mark it as a Streamlit dialog. When the decorated function is called, a dialog element is inserted with the function's body as the content.
+    """Decorator to turn a function into a dialog function which creates a modal overlay when called.
```

Comment:
I'd turn this into "Function decorator to create a modal dialog". That seems a bit easier to read, ties in more nicely with the dialog wording, and the details are explained in the sentence below anyway.